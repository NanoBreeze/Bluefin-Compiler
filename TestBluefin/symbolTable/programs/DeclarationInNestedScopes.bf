int a;
void f() {
	float b;
	string c;
}
bool g(int d) {
	int e;
	{
		{}
		float e;
		int g;
		{
			int g;
		}
	}
	int g;
}