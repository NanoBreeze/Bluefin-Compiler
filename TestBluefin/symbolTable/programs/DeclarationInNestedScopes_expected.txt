resolve - int - builtinType
declare - a - var
resolve - void - builtinType
declare - f - function
enterScope - Level 1
resolve - float - builtinType
declare - b - var
resolve - string - builtinType
declare - c - var
exitScope - Level 1
resolve - bool - builtinType
declare - g - function
enterScope - Level 1
resolve - int - builtinType
declare - d - var
resolve - int - builtinType
declare - e - var
enterScope - Level 2
enterScope - Level 3
exitScope - Level 3
resolve - float - builtinType
declare - e - var
resolve - int - builtinType
declare - g - var
enterScope - Level 3
resolve - int - builtinType
declare - g - var
exitScope - Level 3
exitScope - Level 2
resolve - int - builtinType
declare - g - var
exitScope - Level 1
